.github/workflows/lint-test.yaml

```diff
@@ -60,6 +62,15 @@ jobs:
   # default.ct.yaml
   lint-test:
     runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental }}
```

Comment:
It would be good to keep the main lint-test simple and straightforward. E.g. on the latest stable kind version. Then maybe we can have a separate PR tag `experimental` which triggers an experimental workflow?

Doing it this way with boolean matrix values seems to be putting many concepts into a single workflow which makes it difficult to follow.

Reply:
To elaborate a bit, what this workflow was intended to do is `lint-test` the chart via chart-testing on the stable kind version. This should pick up problems with chart templates, runs on every PR, and is required to merge.

The changes that are being added here are backwards and forwards compatibility changes for various kind versions. I think those should go into a separate workflow which can have experimental flags etc. and probably wouldn't be required to go green before merging.

What do you think?

Reply:
fixed up - moved matrix tests into different workflow that we can mark as non-required.